src/math/lp/nla_core.h

```diff
@@ -118,18 +118,16 @@ class core {
     void insert_to_refine(lpvar j);
     void erase_from_to_refine(lpvar j);
     
-    const lp::u_set&  active_var_set () const { return m_active_var_set;}
+    const indexed_uint_set&  active_var_set () const { return m_active_var_set;}
     bool active_var_set_contains(unsigned j) const { return m_active_var_set.contains(j); }
 
-    void insert_to_active_var_set(unsigned j) const { m_active_var_set.insert(j); }    
+    void insert_to_active_var_set(unsigned j) const { 
+        if (!m_active_var_set.contains(j))
```

Comment:
I created insert_fresh to replace the original insert. Then insert checks containment and is safe. You don't have to wrap the check to contains.